athena-clickhouse/src/main/java/com/amazonaws/athena/connectors/clickhouse/ClickHouseRecordHandler.java

```diff
@@ -0,0 +1,103 @@
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/**
+ * Data handler that reuses {@link MySqlQueryStringBuilder} and {@link MySqlFederationExpressionParser}. user must have necessary permissions to read from necessary tables.
+ */
+public class ClickHouseRecordHandler
+        extends JdbcRecordHandler
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClickHouseRecordHandler.class);
+
+    @VisibleForTesting
+    protected static final String MYSQL_QUOTE_CHARACTER = "`";
+
+    private final JdbcSplitQueryBuilder jdbcSplitQueryBuilder;
+
+    /**
+     * Instantiates handler to be used by Lambda function directly.
+     *
+     * Recommend using {@link MySqlMuxCompositeHandler} instead.
+     */
+    public ClickHouseRecordHandler(java.util.Map<String, String> configOptions)
+    {
+        this(JDBCUtil.getSingleDatabaseConfigFromEnv(ClickHouseConstants.NAME, configOptions), configOptions);
+    }
+
+    public ClickHouseRecordHandler(DatabaseConnectionConfig databaseConnectionConfig, java.util.Map<String, String> configOptions)
+    {
+        this(databaseConnectionConfig, new GenericJdbcConnectionFactory(databaseConnectionConfig, ClickHouseConstants.JDBC_PROPERTIES, new DatabaseConnectionInfo(ClickHouseConstants.DRIVER_CLASS, ClickHouseConstants.DEFAULT_PORT)), configOptions);
+    }
+
+    public ClickHouseRecordHandler(DatabaseConnectionConfig databaseConnectionConfig, JdbcConnectionFactory jdbcConnectionFactory, java.util.Map<String, String> configOptions)
+    {
+        this(databaseConnectionConfig, AmazonS3ClientBuilder.defaultClient(), AWSSecretsManagerClientBuilder.defaultClient(), AmazonAthenaClientBuilder.defaultClient(),
+                jdbcConnectionFactory, new MySqlQueryStringBuilder(MYSQL_QUOTE_CHARACTER, new MySqlFederationExpressionParser(MYSQL_QUOTE_CHARACTER)), configOptions);
+    }
+
+    @VisibleForTesting
+    ClickHouseRecordHandler(DatabaseConnectionConfig databaseConnectionConfig, final AmazonS3 amazonS3, final AWSSecretsManager secretsManager,
+            final AmazonAthena athena, JdbcConnectionFactory jdbcConnectionFactory, JdbcSplitQueryBuilder jdbcSplitQueryBuilder, java.util.Map<String, String> configOptions)
+    {
+        super(amazonS3, secretsManager, athena, databaseConnectionConfig, jdbcConnectionFactory, configOptions);
+        this.jdbcSplitQueryBuilder = Validate.notNull(jdbcSplitQueryBuilder, "query builder must not be null");
+    }
+
+    @Override
+    public PreparedStatement buildSplitSql(Connection jdbcConnection, String catalogName, TableName tableName, Schema schema, Constraints constraints, Split split)
+            throws SQLException
+    {
+        PreparedStatement preparedStatement = jdbcSplitQueryBuilder.buildSql(jdbcConnection, null, tableName.getSchemaName(), tableName.getTableName(), schema, constraints, split);
+
+        // Disable fetching all rows.
+        preparedStatement.setFetchSize(Integer.MIN_VALUE);
```

Comment:
Why are we disabling fetch all rows?

Reply:
Do we want to retrieve it row by row?  per Doc:https://dev.mysql.com/doc/connector-j/en/connector-j-reference-implementation-notes.html

```
To enable this functionality, create a Statement instance in the following manner:

stmt = conn.createStatement(java.sql.ResultSet.TYPE_FORWARD_ONLY,
              java.sql.ResultSet.CONCUR_READ_ONLY);
stmt.setFetchSize(Integer.MIN_VALUE);
The combination of a forward-only, read-only result set, with a fetch size of Integer.MIN_VALUE serves as a signal to the driver to stream result sets row-by-row. After this, any result sets created with the statement will be retrieved row-by-row.
```

Reply:
Correct.  I modeled the code on what I observed in the MySQL connector itself as well as other similar relational ones (e.g. PostgreSQL, RedShift, Oracle, etc).